src/backend/distributed/commands/foreign_data_wrapper.c

```diff
@@ -0,0 +1,135 @@
+/*-------------------------------------------------------------------------
+ *
+ * foreign_data_wrapper.c
+ *    Commands for FOREIGN DATA WRAPPER statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/pg_foreign_data_wrapper.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/commands.h"
+#include "distributed/deparser.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata/distobject.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/worker_transaction.h"
+#include "foreign/foreign.h"
+#include "nodes/primnodes.h"
+
+List *
+PreprocessCreateFdwStmt(Node *node, const char *queryString,
+						ProcessUtilityContext processUtilityContext)
+{
+	EnsureCoordinator();
```

Comment:
EnsureCoordinator should be below ShouldPropagate, otherwise we block this command on workers altogether

Reply:
ah, makes sense